src/core/CONVENTIONS.md

```diff
@@ -30,7 +30,7 @@ my_plugin/
     ├── index.ts
     └── plugin.ts
 ```
-
+- `my_plugin` should be the same as plugin `id` declared in `kibana.json` [manifest file](/docs/development/core/server/kibana-plugin-server.pluginmanifest.md)
```

Comment:
Not sure about this. want to hear your opinion.

Reply:
I think it definitely makes sense. We might even want to check it during plugin discovery and throw if that's not the case.

Reply:
We can try and see if it breaks something. There is a problem that we enforce snake_case in the directory name and now this requirement leaks to `id`.